src/bin/lpython.cpp

```diff
@@ -269,55 +268,84 @@ int emit_llvm(const std::string &infile,
     LFortran::diag::Diagnostics diagnostics;
     LFortran::LocationManager lm;
+    std::vector <std::pair<std::string, double>> times;
+    
+    auto start_file = std::chrono::high_resolution_clock::now();
     lm.in_filename = infile;
     std::string input = LFortran::read_file(infile);
     lm.init_simple(input);
-    LFortran::Result<LFortran::LPython::AST::ast_t*> r = parse_python_file(
-        al, runtime_library_dir, infile, diagnostics, compiler_options.new_parser);
+    auto end_file = std::chrono::high_resolution_clock::now();
+    times.push_back(std::make_pair("Reading_file", std::chrono::duration<double, std::milli>(end_file - start_file).count()));
+
+    auto start_parser = std::chrono::high_resolution_clock::now();
+    LFortran::Result < LFortran::LPython::AST::ast_t * > r = parse_python_file(
+            al, runtime_library_dir, infile, diagnostics, compiler_options.new_parser);
+    auto end_parser = std::chrono::high_resolution_clock::now();
+    times.push_back(std::make_pair("parsing", std::chrono::duration<double, std::milli>(end_parser - start_parser).count()));
     std::cerr << diagnostics.render(input, lm, compiler_options);
     if (!r.ok) {
+        print_report(times, time_report);
         return 1;
     }
 
     // Src -> AST -> ASR
-    LFortran::LPython::AST::ast_t* ast = r.result;
+    auto start_src_to_ast = std::chrono::high_resolution_clock::now();
+    LFortran::LPython::AST::ast_t *ast = r.result;
     diagnostics.diagnostics.clear();
-    LFortran::Result<LFortran::ASR::TranslationUnit_t*>
-        r1 = LFortran::LPython::python_ast_to_asr(al, *ast, diagnostics, true,
-            compiler_options.symtab_only);
+    LFortran::Result < LFortran::ASR::TranslationUnit_t * >
+    r1 = LFortran::LPython::python_ast_to_asr(al, *ast, diagnostics, true,
+                                              compiler_options.symtab_only);
+    auto end_src_to_ast = std::chrono::high_resolution_clock::now();
+    times.push_back(std::make_pair("AST -> ASR", std::chrono::duration<double, std::milli>(end_src_to_ast - start_src_to_ast).count()));
     std::cerr << diagnostics.render(input, lm, compiler_options);
     if (!r1.ok) {
+        print_report(times, time_report);
         LFORTRAN_ASSERT(diagnostics.has_error())
         return 2;
     }
-    LFortran::ASR::TranslationUnit_t* asr = r1.result;
+    LFortran::ASR::TranslationUnit_t *asr = r1.result;
     diagnostics.diagnostics.clear();
 
     // ASR -> LLVM
+    auto start_asr_to_llvm = std::chrono::high_resolution_clock::now();
     LFortran::PythonCompiler fe(compiler_options);
     LFortran::LLVMEvaluator e(compiler_options.target);
-    std::unique_ptr<LFortran::LLVMModule> m;
-    LFortran::Result<std::unique_ptr<LFortran::LLVMModule>>
-        res = fe.get_llvm3(*asr, diagnostics);
+    std::unique_ptr <LFortran::LLVMModule> m;
+    LFortran::Result <std::unique_ptr<LFortran::LLVMModule>>
+            res = fe.get_llvm3(*asr, diagnostics);
```

Comment:
The above seem like just formatting changes. I would do those as either separate PR, or at least separate commits. I think it looks better if written as `std::unique_ptr<LFortran::LLVMModule>` rather than `std::unique_ptr <LFortran::LLVMModule>`, doesn't it?

Reply:
i made this formate by ide formatter, but i change it no problem.

Reply:
Which formatter do you use? We should probably setup a `clang-format` config file for the project.

Reply:
it just a formatter with clion ide with no standard i think, but i will add clang to be the formatter.